From 40ce287776bb8f7917bd4225d4f4ed3a40c25379 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <michael.i.doherty@intel.com>
Date: Thu, 5 Feb 2015 17:03:54 +0000
Subject: [PATCH 3/3] Support includeoptdir, for non fatal inclusion of
 directories

Signed-off-by: Ikey Doherty <michael.i.doherty@intel.com>
---
 mysys/my_default.c | 38 ++++++++++++++++++++++++++++++++++++++
 1 file changed, 38 insertions(+)

diff --git a/mysys/my_default.c b/mysys/my_default.c
index b06e68d..f179657 100644
--- a/mysys/my_default.c
+++ b/mysys/my_default.c
@@ -758,6 +758,7 @@ static int search_default_file_with_ext(Process_option_func opt_handler,
   char name[FN_REFLEN + 10], buff[4096], curr_gr[4096], *ptr, *end, **tmp_ext;
   char *value, option[4096+2], tmp[FN_REFLEN];
   static const char includedir_keyword[]= "includedir";
+  static const char includeoptdir_keyword[] = "includeoptdir";
   static const char include_keyword[]= "include";
   const int max_recursion_level= 10;
   MYSQL_FILE *fp;
@@ -870,6 +871,42 @@ static int search_default_file_with_ext(Process_option_func opt_handler,
 
         my_dirend(search_dir);
       }
+      else if ((!strncmp(ptr, includeoptdir_keyword, sizeof(includeoptdir_keyword) - 1)) &&
+               my_isspace(&my_charset_latin1, ptr[sizeof(includeoptdir_keyword)-1]))
+      {
+
+	if (!(ptr= get_argument(includeoptdir_keyword,
+                                sizeof(includeoptdir_keyword),
+                                ptr, name, line)))
+	  goto err;
+
+        if (!(search_dir= my_dir(ptr, MYF(ME_JUST_WARNING|MY_WANT_SORT))))
+          goto nonfatal;
+
+        for (i= 0; i < (uint) search_dir->number_of_files; i++)
+        {
+          search_file= search_dir->dir_entry + i;
+          ext= fn_ext2(search_file->name);
+
+          /* check extension */
+          for (tmp_ext= (char**) f_extensions; *tmp_ext; tmp_ext++)
+          {
+            if (!strcmp(ext, *tmp_ext))
+              break;
+          }
+
+          if (*tmp_ext)
+          {
+            fn_format(tmp, search_file->name, ptr, "",
+                      MY_UNPACK_FILENAME | MY_SAFE_PATH);
+
+            search_default_file_with_ext(opt_handler, handler_ctx, "", "", tmp,
+                                         recursion_level + 1);
+          }
+        }
+
+        my_dirend(search_dir);
+      }
       else if ((!strncmp(ptr, include_keyword, sizeof(include_keyword) - 1)) &&
                my_isspace(&my_charset_latin1, ptr[sizeof(include_keyword)-1]))
       {
@@ -993,6 +1030,7 @@ static int search_default_file_with_ext(Process_option_func opt_handler,
         goto err;
     }
   }
+ nonfatal:
   mysql_file_fclose(fp, MYF(0));
   return(0);
 
-- 
1.9.1

