From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Arjan van de Ven <arjan@linux.intel.com>
Date: Fri, 10 Jan 2020 10:48:25 -0800
Subject: [PATCH] Enable loading the avx plugin where applicable.

---
 sql/sql_plugin.cc | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/sql/sql_plugin.cc b/sql/sql_plugin.cc
index 981420f0..2c87e1c1 100644
--- a/sql/sql_plugin.cc
+++ b/sql/sql_plugin.cc
@@ -714,6 +714,7 @@ static st_plugin_dl *plugin_dl_add(const LEX_CSTRING *dl, myf MyFlags)
 {
 #ifdef HAVE_DLOPEN
   char dlpath[FN_REFLEN];
+  char dlpath2[FN_REFLEN + 8];
   size_t plugin_dir_len,i;
   uint dummy_errors;
   struct st_plugin_dl *tmp= 0, plugin_dl;
@@ -748,8 +749,18 @@ static st_plugin_dl *plugin_dl_add(const LEX_CSTRING *dl, myf MyFlags)
   strxnmov(dlpath, sizeof(dlpath) - 1, opt_plugin_dir, "/", dl->str, NullS);
   (void) unpack_filename(dlpath, dlpath);
   plugin_dl.ref_count= 1;
+
+  sprintf(dlpath2, "%s.avx2", dlpath);
+
+  plugin_dl.handle= NULL;
+  if (__builtin_cpu_supports("avx2"))
+	  plugin_dl.handle= dlopen(dlpath2, RTLD_NOW);
+
+  if (!(plugin_dl.handle))
+	  plugin_dl.handle= dlopen(dlpath, RTLD_NOW);
+
   /* Open new dll handle */
-  if (!(plugin_dl.handle= dlopen(dlpath, RTLD_NOW)))
+  if (!(plugin_dl.handle))
   {
     my_error(ER_CANT_OPEN_LIBRARY, MyFlags, dlpath, errno, my_dlerror(dlpath));
     goto ret;
