From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Karthik Prabhu Vinod <karthik.prabhu.vinod@intel.com>
Date: Thu, 7 Jan 2021 17:18:07 -0800
Subject: [PATCH] Add Runtime Dependency for libssl & libcrypto

Signed-off-by: Karthik Prabhu Vinod <karthik.prabhu.vinod@intel.com>
---
 support-files/mariadb.pc.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/support-files/mariadb.pc.in b/support-files/mariadb.pc.in
index 500e148d..0b4e47d3 100644
--- a/support-files/mariadb.pc.in
+++ b/support-files/mariadb.pc.in
@@ -20,5 +20,6 @@ Name: @CPACK_PACKAGE_NAME@
 Description: @CPACK_PACKAGE_DESCRIPTION_SUMMARY@
 URL: @CPACK_PACKAGE_URL@
 Version: @VERSION@
+Requires: libssl libcrypto
 Libs: -L${libdir} @LIBS_FOR_CLIENTS@
 Cflags: -I${includedir} @CFLAGS_FOR_CLIENTS@
