--- mariadb-10.3.13/sql/sys_vars.cc~	2019-02-20 07:59:12.000000000 +0000
+++ mariadb-10.3.13/sql/sys_vars.cc	2019-06-06 01:12:57.682101680 +0000
@@ -2831,7 +2831,7 @@
 /* Windows is using OS threadpool, so we're pretty sure it works well */
 #define DEFAULT_THREAD_HANDLING 2
 #else
-#define DEFAULT_THREAD_HANDLING 0
+#define DEFAULT_THREAD_HANDLING 2
 #endif
 
 static Sys_var_enum Sys_thread_handling(
